ServerCore/Pages/CustomHtmlPageLoader.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+
+namespace ServerCore.Pages
+{
+    public class CustomHtmlPageLoader : Controller
+    {
+        private readonly PuzzleServerContext context;
+        private readonly UserManager<IdentityUser> userManager;
+
+        public CustomHtmlPageLoader(PuzzleServerContext context, UserManager<IdentityUser> userManager)
+        {
+            this.context = context;
+            this.userManager = userManager;
+        }
+
+        // GET api/custom
+        [HttpGet]
+        [Route("{eventId}/{puzzleId}/api/custom/")]
+        public async Task<IActionResult> Index(string eventId, int puzzleId)
+        {
+            Event currentEvent = await EventHelper.GetEventFromEventId(context, eventId);

```

Comment:
Can some/all of these checks be replaced by the Authorize attribute with an appropriate policy?